cmd/gameserver/packet/character.go

```diff
@@ -186,3 +186,26 @@ func SetCharacterSlotOrder(session *network.Session, reader *network.Reader) {
 
 	session.Send(packet)
 }
+
+func NewTargetUser(session *network.Session, reader *network.Reader) {
+	sessionId := reader.ReadUint16()
+
+	pSession := g_NetworkManager.GetSession(sessionId)
+	ctx, err := parseSessionContext(pSession)
+	if err != nil {
+		log.Error(err.Error())
+		return
+	}
+
+	var packet = network.NewWriter(NEW_TARGET_USER)
+
+	g_NetworkManager.RLock()
```

Comment:
This network manager Lock/Unlock shouldn't be used here, when reading from the character, you should use it's mutex:

```Go
ctx, err := context.Parse(session)
// handle err

ctx.Mutex.RLock()
current, max := ctx.Char.CurrentHP, ctx.Char.MaxHP
ctx.Mutex.RUnlock()

// create packet and send it
```